### DISCUSSION TITLE
Need help with fireballs

### DISCUSSION THREAD

**Discussion by HierarchyMan** [diamond-rip/Eden#417] [Q&A] [answered]
So I need a plugin that handles 
- fireballs and tnt
- blocks to break in config.yml
- damage and kb modifier

And Eden fireballfight is the bestest perfect Goodest (🤨) example

I'm noob at java so can u tell me
- where the classes are? 
- and possibly even how to compile only the fireball and tnt handler into a single jar with config for the blocks to break damage and kb?

**Comment by GoodestEnglish** [comment, marked as answer]
https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/util/FireballUtil.java - The class which manages the direction of the fireball
https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/match/listener/MatchListener.java#L580C25-L587C57 - The event which handles the fireball shooting
https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/match/listener/MatchListener.java#L349C17-L351C181 - The event which manage the fireball damage and player knockback

**Reply by HierarchyMan** [reply]
And tnt?

**Reply by GoodestEnglish** [reply]
https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/match/listener/MatchListener.java#L777C13-L789C14 - TNT auto ignite when place

**Reply by HierarchyMan** [reply]
ive been trying for so long but cant make it myself
how did i make specific blocks explode with tnt and fireball?

**Reply by GoodestEnglish** [reply]
TNT and fireball have different explode power. 
You can set it by changing the explode entity yield

Something like this:
```java
fireball.setYield(2.0f);
```
